### PR TITLE
add internet radio support

### PR DESCRIPTION
**Pull request by kgarner7**
This PR adds support for internet radios. Specifically:
- It adds an internet radio table
- Implements all subsonic endpoints. Edit endpoints are only possible by admin
- Adds native API support, views
- Adds the ability to play radios in the main app

Limitations (currently):
- Since radio streams have no set length, some browsers will just display X / 0:00 NaN (it appears to be chromium-based ones). This happens even when setting duration
- You can only play (not play next, or later). This is because for some reason I got duplicate key errors in the player.

### CODE REVIEW

**Review by deluan — CHANGES_REQUESTED** [review]
This looks great! Just a few comments below. Also, in terms of UX, I think we need to make the radio clickable from the list, to be consistent with the Songs list.

Also on the List, we should make the HomePageUrl visible by default and also clicakble.

**Comment by kgarner7** [comment]
By clickable, do you mean that clicking on a radio should set dispatch setTrack, and then have view/edit as the menu item? The other changes are easy enough to do.

**Comment by deluan** [comment]
> By clickable, do you mean that clicking on a radio should set dispatch setTrack, and then have view/edit as the menu item? The other changes are easy enough to do.

Maybe make the streamUrl clickable and dispatching setTrack, but everywhere else in the line open the Edit view?

**Comment by deluan** [comment]
I was looking into the react-jinke-player, seems that we will need to set the `visibility` property of the duration element to `hidden` to get rid of the `00:0NaN`. Also, we should set the Player's `seeked` property to false, to disable clicking/dragging the progressbar

File: ui/src/radio/RadioShow.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      {record && <RadioActions record={record} />}
+      {record && (
+        <Card>
+          <SimpleShowLayout>
+            <TextField source="name" validate={[required()]} />
+            <UrlField type="url" source="streamUrl" rel="noreferrer noopener" />
+            <UrlField
+              type="url"
+              source="homePageUrl"
+              rel="noreferrer noopener"
```
Comment (deluan) [inline ui/src/radio/RadioShow.js:31]: We also need to set the `target` attribute to `_blank`

File: ui/src/radio/RadioShow.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  return (
+    <>
+      {record && <RadioActions record={record} />}
+      {record && (
+        <Card>
+          <SimpleShowLayout>
+            <TextField source="name" validate={[required()]} />
+            <UrlField type="url" source="streamUrl" rel="noreferrer noopener" />
```
Comment (deluan) [inline ui/src/radio/RadioShow.js:27]: This should play the radio, instead of opening the stream in the browser.

**Comment by kgarner7** [comment]
I've made the stream url a button that triggers play on press. Hopefully the fact it's a button helps signify that you can interact with it separately. Also made the progress bar invisible if the current song is a radio.

**Comment by deluan** [comment]
I did some changes, and added a responsive Radio list for mobile. The issue is that I was not able to stop the click propagation on the SimpleList, so when I click the "play" button, it also opens the Edit view. Any ideas?

**Comment by kgarner7** [comment]
It looks like if you call `event.preventDefault()` on both the button and icon, then it will work (not sure why it's necessary on both components but /shrug/)

File: ui/src/index.css [hunk 3]
```diff
@@ -11,3 +11,8 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
     monospace;
 }
+
+.radio-player .audio-main,
```
Comment (deluan) [inline ui/src/index.css:15]: Is there a way to move this to scoped styles (with `makeStyles`)? I don't like adding global styles to the app like this, unless strictly necessary

**Review by deluan — COMMENTED** [review]
LGTM. Just a small nit pick, can you take a look please?

**Review by deluan — APPROVED** [review]

**Pull request merged**